bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -3061,7 +3079,9 @@ void RewriteInstance::selectFunctionsToProcess() {
     if (Function.isFragment())
       continue;
 
-    if (!shouldProcess(Function)) {
+    if (isStaticInitFunction(*BC, Function)) {
+      Function.setIgnored();
```

Comment:
Consider using `BC.getBinaryDataByName` to make this a one-time lookup.

Reply:
Done in [80b41c2](https://github.com/llvm/llvm-project/pull/117751/commits/80b41c268c5cbc3feb8b143299140b121d5bad19).